api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -212,7 +212,9 @@ public async Task<IActionResult> Preview(int id)
         var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
         var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
         var report = _reportService.FindById(id) ?? throw new InvalidOperationException("Report does not exist");
-        if (report.OwnerId != user?.Id && !report.IsPublic) throw new NotAuthorizedException("Not authorized to preview this report");
+        if (report.OwnerId != user.Id && // User does not own the report
+            !report.SubscribersManyToMany.Any(s => s.IsSubscribed && s.UserId == user.Id) &&  // User is not subscribed to the report
```

Comment:
@jdtoombs We will now allow a user who is subscribed to a report the ability to preview.